server/worker/software_worker.go

```diff
@@ -0,0 +1,185 @@
+package worker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/mdm/android"
+	kitlog "github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"google.golang.org/api/androidmanagement/v1"
+)
+
+const SoftwareWorkerJobName = "software_worker"
+
+type SoftwareWorkerTask string
+
+type SoftwareWorker struct {
```

Comment:
Would be great to have a test for this worker (you can look at other workers for test mocks and inspiration).